### ISSUE TITLE
Autosave fails if another process is using the autosave

### ISSUE THREAD

**Issue by robmcmullen** [robmcmullen/peppy#637]
_[Trac time 20090518 000811Z]_ On windows, trying to create a new autosave file breaks if another process is using the same autosave:

```
Traceback (most recent call last):
  File "/home/rob/src/peppy-multikey/peppy/buffers.py", line 579, in OnChanged
  File "/home/rob/src/peppy-multikey/peppy/buffers.py", line 612, in removeAutosaveIfExists
  File "/home/rob/src/peppy-multikey/peppy/vfs/itools/vfs/vfs.py", line 89, in remove
  File "/home/rob/src/peppy-multikey/peppy/vfs/itools/vfs/folders.py", line 130, in remove
  File "/home/rob/src/peppy-multikey/peppy/vfs/itools/vfs/file.py", line 184, in remove
WindowsError: [Error 32] The process cannot access the file because it is being used by another process: u'z:/src/peppy/tests/samples/#job-control.py#'
```

**Comment by robmcmullen** [comment]
_[Trac time 20091218 200800Z]_ This also appears to occur if the peppy process is killed via Ctrl-C and then immediately restarted trying to edit the same file

**Comment by robmcmullen** [comment]
_[Trac time 20091220 063312Z]_ f13b408b1baf3c9f5695a68e97d39ba7cbad36d1 Refs #637: added test to see if autosave file removal raised an error
- added autosave_valid instance attribute to Buffer

**Comment by robmcmullen** [comment]
_[Trac time 20091220 063315Z]_ cbf74c2dd5deee583a33710618e1d6d227d825a0 Fixed #637: Fixed hang that was apparent on windows where removeAutosaveIfExists was getting called a huge number of times
- added removeDocumentChangeEvent to stc interface
- added add/removeDocumentChangeEvent during the revert process so extraneous change events aren't generated during the revert
